mutex.go

```diff
@@ -56,7 +56,17 @@ func (m *Mutex) LockContext(ctx context.Context) error {
 
 	for i := 0; i < m.tries; i++ {
 		if i != 0 {
-			time.Sleep(m.delayFunc(i))
+			select {
```

Comment:
btw just noticed this, and i think it could be related to an issue i'm having, but i haven't actually tested this change, just made the edits in the github UI lol.. let me know if it's something you'd accept and i can put some more work into it, thanks!

Reply:
Since `i != 0` here, doesn't that mean that the function will try to acquire the lock at least once? Shouldn't `ctx` be also passed onto the Redis clients if we really want to make use of contexts?

Reply:
Yeah so the context gets passed to the `acquire()` function for the redis calls, so those calls will get cancelled correctly. There may be some need for `actOnPoolsAsync()` to take a context so it can cancel early, but I haven't really thought that through.

This change is mostly just for making sure the `LockContext()` call isn't stuck in a blocking `time.Sleep()` when it should really be cancelled early.